openedx/frontend-app-course-authoring
# feat: customize the certificate link in header

## Description

This PR adds functionality to customize the inclusion of the certificate page link in the header settings menu. When the `CERTIFICATES_HTML_VIEW` feature flag is turned off, the certificate link is removed from the legacy Studio. However, since it is hardcoded in the Authoring MFE, the link still appears in the header settings menu, leading to a 404 page if `CERTIFICATES_HTML_VIEW` is turned off. This PR introduces an environment variable to customize this behavior.

Useful information to include:
- Which edX user roles will this change impact? "Course Author"
- Include screenshots for changes to the UI (ideally, both "before" and "after" screenshots, if applicable).

Before
<img width="432" alt="Screenshot 2024-08-21 at 4 00 37 PM" src="https://github.com/user-attachments/assets/ca3c989d-95b9-47c1-a711-94a817a52b36">

After
<img width="432" alt="Screenshot 2024-08-21 at 3 48 30 PM" src="https://github.com/user-attachments/assets/c1f5ea47-8b01-4b6e-9269-4c839fbd704d">

## Supporting information

https://github.com/mitodl/hq/issues/4658

## Testing instructions

Note: Make sure you enable the contentstore.new_studio_mfe.use_new_course_outline_page waffle flag so that this MFE will be used for displaying the course outline.

- In Studio, visit any course outline and click the Settings tab. You should see a certificate link there.
- In the .env.development file, set ENABLE_CERTIFICATE_PAGE=false.
- Restart the MFE container. Now, repeat the first step. There should be no certificate link.


## Other information

Include anything else that will help reviewers and consumers understand the change.
- Does this change depend on other changes elsewhere?
- Any special concerns or limitations? For example: deprecations, migrations, security, or accessibility.
